File: algebra-perm2a/js/RANDOMGEN_PRIMOS.js
const RANDOMBITS = require('./RANDOMBITS');
const MILLER_RABIN = require('./MILLER_RABIN');

module.exports = b => {
  let s = 0;
  let n = RANDOMBITS(b);

  while (!MILLER_RABIN(n, s))
    n += 2
  
  return n;
}

File: algebra-perm2a/cpp/RANDOMGEN_PRIMOS.cpp
#include "RANDOMBITS.cpp"

#ifndef MILLER_RABIN_F_
#include "MILLER_RABIN.cpp"
#define MILLER_RABIN_F_
#endif

int RANDOMGEN_PRIMOS(int b) {
  int s = 0;
  int n = RANDOMBITS(b);

  while (!MILLER_RABIN(n, s))
    n += 2;
  
  return n;
}
